packages/modules/providers/auth-emailpass/src/services/emailpass.ts

```diff
@@ -43,11 +43,11 @@ export class EmailPassAuthService extends AbstractAuthModuleProvider {
   }
 
   async update(
-    data: { password: string; entity_id: string },
+    data: { password: string; email: string },
```

Comment:
In the docs we mention that we pass `email` and `password`, but in the `auth/<actor_type>/emailpass/update` route we pass the `entity_id` as well that is the `actor_id`. 

Can we please make sure we are not breaking that endpoint with this?

Reply:
@olivermrbl I am not sure how this worked, but the `actor_id` is eg. user_123456, and we passed this as an entity ID, but in the authModule `entity_id` is expected to be the provider's ID eg. `email`.

I think we need to first filter on auth_identity_id, and only after on entity_id which is eg. the email.

Reply:
@sradevski, we currently set the actor ID as the entity ID in the middleware, so there is a slight inconsistency here that we can fix. But this means the entity IDfrom the token is actually passed.

Reply:
> I think we need to first filter on auth_identity_id, and only after on entity_id which is eg. the email.

Also, not sure I understand this part. Where do you propose we add this?

Reply:
@olivermrbl 

There are two approaches I am thinking of.

**The first one is:**
We pass the authIdentityId to updateProvider, and then to the method factory (it would be optional here)
```

  async updateProvider(
    provider: string,
    authIdentityId: string,
    data: Record<string, unknown>
  ): Promise<AuthenticationResponse> {
    try {
      return await this.authProviderService_.update(
        provider,
        data,
        this.getAuthIdentityProviderService(provider, authIdentityId)
      )
    } catch (error) {
      return { success: false, error: error.message }
    }
  }

```

Then, in the method factory we essentially apply the auth_identity_id to all methods there (list, retrieve), scoping the requests to this specific auth identity ID.

**The second approach is to deletgate these checks to the caller:**
In the HTTP layer we call listAuthIdentities with the auth identity ID from the context, and we check if the provider being updated is part of the `providerIdentities.entity_id` for that authIdentityId. If it is, call updateProvider as now, if not reject.

Reply:
> In the HTTP layer we call listAuthIdentities with the auth identity ID from the context, and we check if the provider being updated is part of the providerIdentities.entity_id for that authIdentityId. If it is, call updateProvider as now, if not reject.

This is essentially what we do right now, when a token is passed to the endpoint. Bear in mind we haven't yet added support for updating your provider without a token. We can add that later down the line when we want to support more use cases than password resets.

In the token validation middleware, we pull the `entity_id` from the token and list auth identities with `(provider, entity_id)`. If the auth identity doesn't exist, we reject the request.

Reply:
@olivermrbl thanks for the clarification, Will cancel the pr